pcdsdevices/analog_signals.py

```diff
@@ -0,0 +1,167 @@
+import time
+import pcdsdevices.utils as key_press
+from ophyd import Device, Component as Cpt, EpicsSignal, EpicsSignalRO
+
+
+class Acromag(Device):
+    """
+    Class for Acromag analog input/ouput signals
+
+    Parameters:
+    -----------
+    prefix : str
+        The Epics base of the acromag
+
+    name : str
+        A name to prefer to the device
+    """
+    # Components for each channel
+    ao1_0 = Cpt(EpicsSignal, ":ao1:0", kind='normal')
+    ao1_1 = Cpt(EpicsSignal, ":ao1:1", kind='normal')
+    ao1_2 = Cpt(EpicsSignal, ":ao1:2", kind='normal')
+    ao1_3 = Cpt(EpicsSignal, ":ao1:3", kind='normal')
+    ao1_4 = Cpt(EpicsSignal, ":ao1:4", kind='normal')
+    ao1_5 = Cpt(EpicsSignal, ":ao1:5", kind='normal')
+    ao1_6 = Cpt(EpicsSignal, ":ao1:6", kind='normal')
+    ao1_7 = Cpt(EpicsSignal, ":ao1:7", kind='normal')
+    ao1_8 = Cpt(EpicsSignal, ":ao1:8", kind='normal')
+    ao1_9 = Cpt(EpicsSignal, ":ao1:9", kind='normal')
+    ao1_10 = Cpt(EpicsSignal, ":ao1:10", kind='normal')
+    ao1_11 = Cpt(EpicsSignal, ":ao1:11", kind='normal')
+    ao1_12 = Cpt(EpicsSignal, ":ao1:12", kind='normal')
+    ao1_13 = Cpt(EpicsSignal, ":ao1:13", kind='normal')
+    ao1_14 = Cpt(EpicsSignal, ":ao1:14", kind='normal')
+    ao1_15 = Cpt(EpicsSignal, ":ao1:15", kind='normal')
+
+    ai1_0 = Cpt(EpicsSignalRO, ":ai1:0", kind='normal')
+    ai1_1 = Cpt(EpicsSignalRO, ":ai1:1", kind='normal')
+    ai1_2 = Cpt(EpicsSignalRO, ":ai1:2", kind='normal')
+    ai1_3 = Cpt(EpicsSignalRO, ":ai1:3", kind='normal')
+    ai1_4 = Cpt(EpicsSignalRO, ":ai1:4", kind='normal')
+    ai1_5 = Cpt(EpicsSignalRO, ":ai1:5", kind='normal')
+    ai1_6 = Cpt(EpicsSignalRO, ":ai1:6", kind='normal')
+    ai1_7 = Cpt(EpicsSignalRO, ":ai1:7", kind='normal')
+    ai1_8 = Cpt(EpicsSignalRO, ":ai1:8", kind='normal')
+    ai1_9 = Cpt(EpicsSignalRO, ":ai1:9", kind='normal')
+    ai1_10 = Cpt(EpicsSignalRO, ":ai1:10", kind='normal')
+    ai1_11 = Cpt(EpicsSignalRO, ":ai1:11", kind='normal')
+    ai1_12 = Cpt(EpicsSignalRO, ":ai1:12", kind='normal')
+    ai1_13 = Cpt(EpicsSignalRO, ":ai1:13", kind='normal')
+    ai1_14 = Cpt(EpicsSignalRO, ":ai1:14", kind='normal')
+    ai1_15 = Cpt(EpicsSignalRO, ":ai1:15", kind='normal')
+
+    def initialize_mesh(self, sp_ch, rb_ch, scale=1000.0):
```

Comment:
I don't think these mesh methods belong on the `Acromag` class. I think these are more suitable for a subclass of `Acromag`. A majority of `Acromag` devices do not have the mesh power supply attached to them.

Reply:
Oh that makes sense to me. I'll move the mesh functions into their own class that inherits from Acromag, that will clean up the way I was doing things as I would no longer need the `initialize_mesh` function. I'll also try to implement @klauer 's suggestion as I think that would clean things up.